six/modules/c++/six.sicd/source/Utilities.cpp

```diff
@@ -784,6 +877,194 @@ std::auto_ptr<ComplexData> Utilities::createFakeComplexData()
+{
+    const std::map<std::string, size_t> nameToDesIndex =
+        getAdditionalDesMap(reader);
+
+    // Slant and output plane mesh IDs must be present in the DES
+    if (nameToDesIndex.find(SICDMeshes::NOISE_MESH_ID) == nameToDesIndex.end())
+    {
+        throw except::Exception(Ctxt("Noise mesh information not present"));
+    }
+
+    // Extract the noise mesh
+    std::auto_ptr<NoiseMesh> noiseMesh(new NoiseMesh(SICDMeshes::NOISE_MESH_ID));
+    mem::ScopedAlignedArray<sys::byte> buffer;
+    getDesBuffer(
+        reader, nameToDesIndex.at(SICDMeshes::NOISE_MESH_ID), buffer);
+    sys::byte* bufferData = buffer.get();
+    noiseMesh->deserialize(bufferData);
+
+    return noiseMesh;
+}
+
+void Utilities::getProjectionPolys(NITFReadControl& reader,
+                                   std::auto_ptr<ComplexData>& complexData,
+                                   six::Poly2D& outputRowColToSlantRow,
+                                   six::Poly2D& outputRowColToSlantCol)
+{
+    const std::map<std::string, size_t> nameToDesIndex =
+        getAdditionalDesMap(reader);
+
+    // Slant and output plane mesh IDs must be present in the DES
+    if (nameToDesIndex.find(SICDMeshes::SLANT_PLANE_MESH_ID) == nameToDesIndex.end())
+    {
+        throw except::Exception(Ctxt("Slant plane mesh information not present"));
+    }
+
+    if (nameToDesIndex.find(SICDMeshes::OUTPUT_PLANE_MESH_ID) == nameToDesIndex.end())
+    {
+        throw except::Exception(Ctxt("Outputplane mesh information not present"));
+    }
+
+    mem::ScopedAlignedArray<sys::byte> buffer;
+    sys::byte* bufferData;
+
+    // Extract the slant plane mesh buffer and deserialize
+    PlanarCoordinateMesh slantMesh(SICDMeshes::SLANT_PLANE_MESH_ID);
+    getDesBuffer(
+        reader, nameToDesIndex.at(SICDMeshes::SLANT_PLANE_MESH_ID), buffer);
+    bufferData = buffer.get();
+    slantMesh.deserialize(bufferData);
+
+    // Extract the output plane mesh buffer and deserialize
+    PlanarCoordinateMesh outputMesh(SICDMeshes::OUTPUT_PLANE_MESH_ID);
+    getDesBuffer(
+        reader, nameToDesIndex.at(SICDMeshes::OUTPUT_PLANE_MESH_ID), buffer);
+    bufferData = buffer.get();
+    outputMesh.deserialize(bufferData);
+
+    const types::RowCol<double> outputSampleSpacing(
+        complexData->radarCollection->area->plane->xDirection->spacing,
+        complexData->radarCollection->area->plane->yDirection->spacing);
+
+    const types::RowCol<double> outputCenter(
+        complexData->radarCollection->area->plane->referencePoint.rowCol.row,
+        complexData->radarCollection->area->plane->referencePoint.rowCol.col);
+
+    const types::RowCol<double> slantSampleSpacing(
+        complexData->grid->row->sampleSpacing,
+        complexData->grid->col->sampleSpacing);
+
+    const types::RowCol<double> slantCenter(
+        complexData->imageData->scpPixel.row,
+        complexData->imageData->scpPixel.col);
+
+    size_t orderX = 4;
```

Comment:
Doesn't seem like this should be hardcoded.

Reply:
Added as an argument to getProjectionPolys() and, as a result, readSicd()